### ISSUE TITLE
The binding between a node and its script is corrupted when the node is instantiated from a custom class, after packing its parent and reloading it

### ISSUE THREAD

**Issue by ghghorbanzade** [godotengine/godot#78360]
### Godot version

3.5 stable

### System information

Windows 10 - NVIDIA GeForce RTX 2080 SUPER [Version 531.29] - Godot 3.5 stable - GLES3

### Issue description

I am encountering an issue with my game scene where nodes instantiated from classes lose their bindings to their scripts and cannot be called after the scene is packed and reloaded. Initially, when I add these nodes to the "Game.tscn" scene using add_child(), everything works fine. However, upon packing and reloading the scene at runtime, the bindings between the class scripts and their corresponding nodes become corrupted. It's worth noting that when I open the packed scene in the editor, the bindings between the children and their instantiated class scripts remain intact. The problem only arises when reloading the packed "Game.tscn" scene at runtime.

### Steps to reproduce

Main setup: There is a node named "Game" in the scene tree under the main node.
Inherited classes: Two classes, "Player" and "Gun," are extended from the Node class.
Instantiation: In the game scene, after the game starts, a node is created from the "Player" class, and multiple gun nodes are created from the "Gun" class.
Packing the scene: When the game scene is saved using the "PackScene()" function, the "Game" node becomes the parent, and the player and gun nodes become its children in the scene tree.
Loading the saved scene: In another node, the saved "Game.tscn" is loaded.
Binding loss: However, the bindings between the player and gun nodes and their corresponding scripts are lost during the loading process.

### Minimal reproduction project

-

**Comment by Calinou** [comment]
@ghghorbanzade Please upload a [minimal reproduction project](https://github.com/godotengine/godot/blob/master/CONTRIBUTING.md#provide-a-simple-example-project) to make this easier to troubleshoot.

**Comment by ghghorbanzade** [comment]
Thanks, @Calinou. I found out where the problem was caused. I used custom _init() functions in my defined classes (Player and Gun), which need some properties to pass to the _init() functions. It seems that the necessary properties to pass into the _init() function can't be passed in reloading the saved scene. So, the binding of nodes that are instantiated from classes is lost to the class script. 
I addressed my need to pass input arguments to the classes used by defining a non-virtual method called _initialize().
